classes/statistics/MetricsDAO.inc.php

```diff
@@ -94,9 +94,241 @@ protected function getAssocObjectInfo($submissionId, $contextId) {
 		$publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
 		$publishedArticle = $publishedArticleDao->getByArticleId($submissionId, $contextId, true);
 		if ($publishedArticle) {
-			$returnArray = array(ASSOC_TYPE_ISSUE, $publishedArticle->getIssueId());
+			$returnArray = [ASSOC_TYPE_ISSUE, $publishedArticle->getIssueId()];
 		}
 		return $returnArray;
 	}
-}
 
+	/**
+	 * Retrieves several statistics about submissions, with sums and averages
+	 * @param $contextId int The context id
+	 * @param $dateStart string The start date filter
+	 * @param $dateEnd string The end date filter
+	 * @param $sectionIds array A list of section IDs to filter the result set
+	 * @return array Returns a dictionary with statistics
+	 */
+	public function getSubmissionStatistics($contextId = null, $dateStart = null, $dateEnd = null, $sectionIds = null) {
+		// Access submission constants.
+		import('lib.pkp.classes.submission.Submission');
+
+		// Access decision actions constants.
+		import('classes.workflow.EditorDecisionActionsManager');
+
+		$params = $where = $whereMinimum = [];
+		if($dateStart || $dateEnd) {
+			$where[] = 'ed.date_decided ' . ($dateStart && $dateEnd ? 'BETWEEN ? AND ?' : ($dateStart ? '>= ?' : '<= ?'));
+			$dateStart && $params[] = $dateStart;
```

Comment:
I need to check what framework you've adopted to handle databases... To avoid being too slow, I just followed the standard, but I'd like to use at least named parameters here.

Reply:
For now, the DAOs use ADODB and the service classes use Illuminate. We're gradually trying to phase out the ADODB-based code in favour of the Illuminate-based code, but don't worry for now about using ADODB for new code in DAOs.